### PR TITLE
release notes 1.24.xx

### PR DESCRIPTION
**Pull request by SushrutHarness**
# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the CODEOWNERS. 

## What Type of PR is This?

- [ ] Issue
- [ ] Feature
- [X ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ X] Tested Locally
- [ ] *Optional* Screenshot.

### CODE REVIEW

File: release-notes/continuous-delivery.md [hunk 1]
```diff
@@ -49,6 +49,32 @@ import Kustomizedep from '/release-notes/shared/kustomize-3-4-5-deprecation-noti
...
+- Fixed an issue where a Targeted Hosts field was not populated in the case of execution input.  (CDS-91071)
+- Fixed an issue where the pipeline failed to execute when remote infrastructure was used with a multi environment deployment. (CDS-90985, ZD-57420)
+- Fixed an issue where users couldn’t switch versions for account level templates. (CDS-89602, ZD-57282, ZD-57296, ZD-57320)
+- The CDK Deploy step was not handling the null output map when the provided image was not created from the harness aws-cdk-plugin base image: https://hub.docker.com/r/harness/aws-cdk-plugin/tags. This issue has been resolved, the null output map is handled. (CDS-89569)
+- Provided new expression version (CDS-89391)
+- Fixed issue with edit file store in window (CDS-89094)
+- The email step body used to render the field's HTML content. Now it will render raw HTML. (CDS-88842, ZD-56452)
+- arlier the input sets that have nested components (such as templates) hosted on GitHub were taking a long time to fetch. Now this time has been significantly reduced. (CDS-88426, ZD-56180)
```
Comment (brian-f-harness) [inline release-notes/continuous-delivery.md:73]:
```suggestion
- Earlier, the input sets that have nested components (such as templates) hosted on GitHub were taking a long time to fetch. Now this time has been significantly reduced. (CDS-88426, ZD-56180)
```

File: release-notes/continuous-delivery.md [hunk 2]
```diff
@@ -49,6 +49,32 @@ import Kustomizedep from '/release-notes/shared/kustomize-3-4-5-deprecation-noti
...
+This results in no deployment even though the pipeline is successful. This issue has not been resolved. Instead we scale down only DaemonSets and delete deployment, deploymentConfig, HPA, PDB and statefulset resources. (CDS-88999, ZD-56645)
+
+#### Fixed Issues
+- The Input Set Page breaks while editing the input set when the service input fields have been updated. Now the page will render with the input set yaml. (CDS-91095, ZD-57487)
+- Fixed an issue where a Targeted Hosts field was not populated in the case of execution input.  (CDS-91071)
+- Fixed an issue where the pipeline failed to execute when remote infrastructure was used with a multi environment deployment. (CDS-90985, ZD-57420)
+- Fixed an issue where users couldn’t switch versions for account level templates. (CDS-89602, ZD-57282, ZD-57296, ZD-57320)
+- The CDK Deploy step was not handling the null output map when the provided image was not created from the harness aws-cdk-plugin base image: https://hub.docker.com/r/harness/aws-cdk-plugin/tags. This issue has been resolved, the null output map is handled. (CDS-89569)
+- Provided new expression version (CDS-89391)
```
Comment (brian-f-harness) [inline release-notes/continuous-delivery.md:70]:
```suggestion
- Provided new expression version. (CDS-89391)
```

Comment (brian-f-harness) [inline release-notes/continuous-delivery.md:70]: Are there more details/additional context for this RN?

File: release-notes/continuous-delivery.md [hunk 3]
```diff
@@ -49,6 +49,32 @@ import Kustomizedep from '/release-notes/shared/kustomize-3-4-5-deprecation-noti
...
+- Fixed issue with edit file store in window (CDS-89094)
```
Comment (brian-f-harness) [inline release-notes/continuous-delivery.md:71]:
```suggestion
- Fixed issue with edit file store in window. (CDS-89094)
```

Comment (brian-f-harness) [inline release-notes/continuous-delivery.md:71]: in Windows? or in a specific window? Recommend updating.

File: release-notes/continuous-delivery.md [hunk 4]
```diff
@@ -49,6 +49,32 @@ import Kustomizedep from '/release-notes/shared/kustomize-3-4-5-deprecation-noti
...
+- If a feature flag is turned on for a module it will now appear in the module selector for the new navigation experience. (CDS-85185)
+
+#### Behavior Changes
+- In the blue/green stage scale down step, we used to scale down deployments, statefulsets, daemonsets , deploymentConfig and delete HPA, and PDB resources. During scale down we updated the field "replicas" to 0. In Kubernetes if HPA is configured it is not mandatory to define replicas. 
+So when another deployment happens and we apply the same old deployments manifest it does not update the replicas field and it remains set to 0. 
+This results in no deployment even though the pipeline is successful. This issue has not been resolved. Instead we scale down only DaemonSets and delete deployment, deploymentConfig, HPA, PDB and statefulset resources. (CDS-88999, ZD-56645)
+
+#### Fixed Issues
+- The Input Set Page breaks while editing the input set when the service input fields have been updated. Now the page will render with the input set yaml. (CDS-91095, ZD-57487)
```
Comment (brian-f-harness) [inline release-notes/continuous-delivery.md:65]:
```suggestion
- The Input Set Page breaks while editing the input set when the service input fields have been updated. Now the page will render with the input set YAML. (CDS-91095, ZD-57487)
```

File: release-notes/continuous-delivery.md [hunk 5]
```diff
@@ -49,6 +49,32 @@ import Kustomizedep from '/release-notes/shared/kustomize-3-4-5-deprecation-noti
...
 
+### Version 1.24.7
+
+#### New Features and Enhancements
+- We've introduced a new expression that explicitly provides the trigger name used to initiate the execution. You can now utilize `<+pipeline.triggeredBy.triggerDisplayName>` to access this information. (CDS-87696)
+- We enhanced the resolution of the working directory. Now you can determine the working directory based on environment variables. This includes variables provided in shell script steps through input variables, as well as those from the host's environment variables. (CDS-87446)
+- If a feature flag is turned on for a module it will now appear in the module selector for the new navigation experience. (CDS-85185)
+
+#### Behavior Changes
+- In the blue/green stage scale down step, we used to scale down deployments, statefulsets, daemonsets , deploymentConfig and delete HPA, and PDB resources. During scale down we updated the field "replicas" to 0. In Kubernetes if HPA is configured it is not mandatory to define replicas.
```
Comment (brian-f-harness) [inline release-notes/continuous-delivery.md:60]:
```suggestion
- In the blue/green stage scale down step, we used to scale down deployments, statefulsets, daemonsets, deploymentConfig and delete HPA, and PDB resources. During scale down, we updated the field `replicas` to 0. In Kubernetes, if HPA is configured it is not mandatory to define replicas. 
```

File: release-notes/continuous-delivery.md [hunk 6]
```diff
@@ -49,6 +49,32 @@ import Kustomizedep from '/release-notes/shared/kustomize-3-4-5-deprecation-noti
...
+- We enhanced the resolution of the working directory. Now you can determine the working directory based on environment variables. This includes variables provided in shell script steps through input variables, as well as those from the host's environment variables. (CDS-87446)
+- If a feature flag is turned on for a module it will now appear in the module selector for the new navigation experience. (CDS-85185)
+
+#### Behavior Changes
+- In the blue/green stage scale down step, we used to scale down deployments, statefulsets, daemonsets , deploymentConfig and delete HPA, and PDB resources. During scale down we updated the field "replicas" to 0. In Kubernetes if HPA is configured it is not mandatory to define replicas. 
+So when another deployment happens and we apply the same old deployments manifest it does not update the replicas field and it remains set to 0. 
+This results in no deployment even though the pipeline is successful. This issue has not been resolved. Instead we scale down only DaemonSets and delete deployment, deploymentConfig, HPA, PDB and statefulset resources. (CDS-88999, ZD-56645)
```
Comment (brian-f-harness) [inline release-notes/continuous-delivery.md:62]:
```suggestion
This results in no deployment even though the pipeline is successful. This issue has not been resolved. Instead, we scale down only DaemonSets and delete deployment, deploymentConfig, HPA, PDB, and statefulset resources. (CDS-88999, ZD-56645)
```

File: release-notes/continuous-delivery.md [hunk 7]
```diff
@@ -49,6 +49,32 @@ import Kustomizedep from '/release-notes/shared/kustomize-3-4-5-deprecation-noti
...
+
+#### Fixed Issues
+- The Input Set Page breaks while editing the input set when the service input fields have been updated. Now the page will render with the input set yaml. (CDS-91095, ZD-57487)
+- Fixed an issue where a Targeted Hosts field was not populated in the case of execution input.  (CDS-91071)
+- Fixed an issue where the pipeline failed to execute when remote infrastructure was used with a multi environment deployment. (CDS-90985, ZD-57420)
+- Fixed an issue where users couldn’t switch versions for account level templates. (CDS-89602, ZD-57282, ZD-57296, ZD-57320)
```
Comment (brian-f-harness) [inline release-notes/continuous-delivery.md:68]:
```suggestion
- Fixed an issue where users couldn’t switch versions for account-level templates. (CDS-89602, ZD-57282, ZD-57296, ZD-57320)
```

File: release-notes/continuous-delivery.md [hunk 8]
```diff
@@ -49,6 +49,32 @@ import Kustomizedep from '/release-notes/shared/kustomize-3-4-5-deprecation-noti
...
+- The CDK Deploy step was not handling the null output map when the provided image was not created from the harness aws-cdk-plugin base image: https://hub.docker.com/r/harness/aws-cdk-plugin/tags. This issue has been resolved, the null output map is handled. (CDS-89569)
```
Comment (brian-f-harness) [inline release-notes/continuous-delivery.md:69]:
```suggestion
- The CDK Deploy step was not handling the null output map when the provided image was not created from the Harness aws-cdk-plugin base image: https://hub.docker.com/r/harness/aws-cdk-plugin/tags. This issue has been resolved, the null output map is handled. (CDS-89569)
```

File: release-notes/continuous-delivery.md [hunk 9]
```diff
@@ -49,6 +49,32 @@ import Kustomizedep from '/release-notes/shared/kustomize-3-4-5-deprecation-noti
...
+
+#### Behavior Changes
+- In the blue/green stage scale down step, we used to scale down deployments, statefulsets, daemonsets , deploymentConfig and delete HPA, and PDB resources. During scale down we updated the field "replicas" to 0. In Kubernetes if HPA is configured it is not mandatory to define replicas. 
+So when another deployment happens and we apply the same old deployments manifest it does not update the replicas field and it remains set to 0. 
+This results in no deployment even though the pipeline is successful. This issue has not been resolved. Instead we scale down only DaemonSets and delete deployment, deploymentConfig, HPA, PDB and statefulset resources. (CDS-88999, ZD-56645)
+
+#### Fixed Issues
+- The Input Set Page breaks while editing the input set when the service input fields have been updated. Now the page will render with the input set yaml. (CDS-91095, ZD-57487)
+- Fixed an issue where a Targeted Hosts field was not populated in the case of execution input.  (CDS-91071)
+- Fixed an issue where the pipeline failed to execute when remote infrastructure was used with a multi environment deployment. (CDS-90985, ZD-57420)
```
Comment (brian-f-harness) [inline release-notes/continuous-delivery.md:67]:
```suggestion
- Fixed an issue where the pipeline failed to execute when remote infrastructure was used with a multi-environment deployment. (CDS-90985, ZD-57420)
```

**Review by brian-f-harness — COMMENTED** [review]
Added some suggested changes.

File: release-notes/continuous-delivery.md (hunk 2 above)
Comment (SushrutHarness) [reply release-notes/continuous-delivery.md:70]: Working on getting those right now, unfortunately the dev didn't really elaborate so looking into what the actual fix was manually.

File: release-notes/continuous-delivery.md (hunk 3 above)
Comment (SushrutHarness) [reply release-notes/continuous-delivery.md]: Same dev and issue as above

**Review by krishi0408 — APPROVED** [review]

**Pull request merged**